spring-kafka/src/test/java/org/springframework/kafka/config/StreamsBuilderFactoryBeanInMemoryStateStoreTests.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2018-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.config;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.KTable;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.annotation.EnableKafkaStreams;
+import org.springframework.kafka.annotation.KafkaStreamsDefaultConfiguration;
+import org.springframework.kafka.test.EmbeddedKafkaBroker;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+/**
+ * @author Cédric Schaller
+ */
+@SpringJUnitConfig
+@DirtiesContext
+@EmbeddedKafka
+public class StreamsBuilderFactoryBeanInMemoryStateStoreTests {
+
+	private static Path stateStoreDir;
+
+	@BeforeAll
+	static void beforeAll() throws IOException {
+		stateStoreDir = Files.createTempDirectory(StreamsBuilderFactoryBeanInMemoryStateStoreTests.class.getSimpleName());
+	}
+
+	@Test
+	void testStateStoreIsInMemory() {
+		assertThat(stateStoreDir).isEmptyDirectory();
```

Comment:
Isn't this always true regardless of the change? Maybe you want to test whether some data was produced in the in-memory state store?

Reply:
No, removing the change will lead to a test failure. 

By default, state stores (and therefore the `state.dir` when using RocksDB) are kept across restarts of the application so that the state does not have to be reconstructed from the Kafka topic. If you want to remove the state store directory when Kafka Streams is shutting down, you have to pass a `CleanupConfig` to the `StreamsBuilderFactoryBean` with `onStop` set to `true`, as it is the case e.g. in `StreamsBuilderFactoryBeanTests:120` for the test called `testCleanupStreams()`. 

At first, I tried to test the fact that we indeed had an in-memory store, but this involved using `ReflectionUtils`, which in my opinion was suboptimal. So checking the non-existence of a RocksDB directory felt like the cleanest way to do this. However, if you have an idea how to test that an in-memory store is used without using reflection, I am happy to modify this test.